torch/csrc/distributed/rpc/utils.cpp

```diff
@@ -150,6 +156,30 @@ std::unique_ptr<RpcCommandBase> deserializeResponse(
     case MessageType::CLEANUP_AUTOGRAD_CONTEXT_RESP: {
       return autograd::CleanupAutogradContextResp::fromMessage(response);
     }
+    case MessageType::RUN_WITH_PROFILING_RESP: {
+      std::unique_ptr<RpcCommandBase> rpcPtr =
+          autograd::RpcWithProfilingResp::fromMessage(response);
+      RpcCommandBase& rpc = *rpcPtr;
+      auto& rpcWithProfilingResp =
+          static_cast<autograd::RpcWithProfilingResp&>(rpc);
+      // read and reconstruct events
+      // Check if the profiler is enabled
+      auto enabled = torch::autograd::profiler::profilerEnabled();
+      TORCH_CHECK(
+          enabled,
+          "Profiler was expected to be enabled. This can happen in callback "
+          " continutations that run in different threads, and the TLS of the "
+          " profiler was not propagated.");
+      std::vector<torch::autograd::profiler::Event> events =
+          rpcWithProfilingResp.getProfiledEvents();
+      auto fromWorker = rpcWithProfilingResp.fromWorkerId();
+      wrappedMsgType = rpcWithProfilingResp.wrappedMessageType();
+
+      torch::autograd::profiler::addEventList(std::move(events));
```

Comment:
It looks like we add the events to a thread local here. Don't we need to tie these remote events back to RPC that actually invoked these on the remote end?

Reply:
I think for now, we can go with the approach of prepending the key that the original RPC that caused these remote events was profiled this, as per what we discussed in some of the other comments? And yes, the thread local here is not tied back to the original RPC currently but is consolidated along with the rest of the profiled events in a `consolidate()` function. I think in the future, we will want ways to associate these remote events as "child" events of the RPC that invoked them.

Reply:
Are we planning to add the changes for prefixing the original key as part of this PR?

Reply:
we should have node_id associated with an event, is that enough or you mean smth else?